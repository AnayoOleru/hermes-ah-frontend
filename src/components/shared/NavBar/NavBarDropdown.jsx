import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

class NavDropdown extends Component {
  constructor(props) {
    super(props);
    this.state = {
      openDropdown: false,
    };
  }

  componentDidMount() {
    const { getProfile } = this.props;
    getProfile();
  }

  toggleDropdown = () => {
    this.setState(prevState => ({ openDropdown: !prevState.openDropdown }));
  };

  render() {
    const { openDropdown } = this.state;
    const { user } = this.props;
    const { userProfile } = user;
    const { profile } = userProfile;
    return (
      <div>
        <button
          type="button"
          className="nav-drp-btn"
          onClick={this.toggleDropdown}
        >
          <div className="nav-drp-btn-ctn">
            <img
              className="nav-drp-pic"
              src={
                !profile || !profile.image_url
                  ? 'https://res.cloudinary.com/sojidan/image/upload/v1557149927/avatar.png'
                  : profile.image_url
              }
              alt=""
            />
            <i className="fas fa-caret-down" />
            {openDropdown ? (
              <div className="dropdown-content">
                <Link to="/profile">My Profile</Link>
                <Link to="/logout">Logout</Link>
                <Link to="/admin">Admin</Link>
              </div>
            ) : null}
          </div>
        </button>
      </div>
    );
  }
}

NavDropdown.propTypes = {
  getProfile: PropTypes.func.isRequired,
  user: PropTypes.shape().isRequired,
};

export default NavDropdown;
