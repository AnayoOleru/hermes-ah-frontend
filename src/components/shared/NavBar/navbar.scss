.ui.menu {
  height: 135px;
  img.logo {
    width: 184px !important;
    height: 135px;
  }
}