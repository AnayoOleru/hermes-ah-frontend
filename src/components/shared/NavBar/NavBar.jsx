import React, { Component } from 'react';
import { Menu } from 'semantic-ui-react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import logo from '../../../assets/images/logo_transparent.png';
import './navbar.scss';

class NavBar extends Component {
  state = {};

  render() {
    const { navLinks } = this.props;
    return (
      <Menu>
        <Menu.Menu className="logo-cont">
          <Menu.Item>
            <Link to="/">
              <img className="logo" src={logo} alt="logo" size="small" />
            </Link>
          </Menu.Item>
        </Menu.Menu>
        <Menu.Menu className="nav-cont">
          {navLinks.map(child => (
            <Link key={child.text} to={child.link}>
              <Menu.Item className={`navbar-item ${child.className}`}>
                {child.text}
              </Menu.Item>
            </Link>
          ))}
        </Menu.Menu>
      </Menu>
    );
  }
}

NavBar.propTypes = {
  navLinks: propTypes.arrayOf(propTypes.shape).isRequired,
};

export default NavBar;
