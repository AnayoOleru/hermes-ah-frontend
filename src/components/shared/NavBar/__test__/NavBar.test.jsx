import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../NavBar';
import NavContainer from '../NavBarDropdown';

const userProfile = {
  userProfile: {
    profile: {
      first_name: 'jest',
      last_name: 'enzyme',
      image_url: 'url',
    },
  },
};

const initUserProfile = {
  userProfile: {
    profile: {
      first_name: 'jest',
      last_name: 'enzyme',
      image_url: 'nourl',
    },
  },
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ userProfile });

describe('NavBar component', () => {
  it('should match snapshot', () => {
    const wrapper = create(
      <Router>
        <Provider store={store}>
          <NavBar user={initUserProfile} />
        </Provider>
      </Router>
    );
    expect(wrapper.toJSON).toMatchSnapshot();
  });

  it('should render tags', () => {
    const wrapper = shallow(
      <Router>
        <NavBar user={userProfile} />
      </Router>
    );
    expect(wrapper.find('menu'));
    expect(wrapper.find('img'));
    expect(wrapper.find('menu.item'));
  });

  it('should change state', () => {
    localStorage.setItem(
      'token',
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyT2JqIjp7ImlkIjoiNTdjNTE1YTEtODkwZC00MTJmLThjYTEtMGE1Mzk1MTIzZGNhIiwiZW1haWwiOiJhbWVhY2hpY2h1a3NAZ21haWwuY29tIiwiaXNBZG1pbiI6dHJ1ZSwiaXNSZXZpZXdlciI6dHJ1ZSwiaXNBY3RpdmF0ZWQiOnRydWV9LCJpYXQiOjE1NTc3Njg3NjcsImV4cCI6MTU1ODAyNzk2N30.msqGVSZbPrBu8ky7Pa-KauhDnMLGRHG4eas9wlIEEA0'
    );

    const props = {
      user: { userProfile: {} },
      getProfile: jest.fn(),
      getSuggestions: jest.fn(),
    };
    const wrapper = shallow(<NavContainer {...props} />);
    wrapper.instance().toggleDropdown();
    expect(wrapper.state('openDropdown')).toBe(true);
  });
});
