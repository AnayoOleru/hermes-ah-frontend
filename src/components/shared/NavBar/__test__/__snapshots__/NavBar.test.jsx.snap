// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should match snapshot 1`] = `[Function]`;
